src/event.c

```diff
@@ -148,43 +202,69 @@ int event_check_value(knot_event event, knot_value_type current_val,
 			value_type > KNOT_VALUE_TYPE_MAX)
 		return -EINVAL;
 
-	if (is_change_flag_set(event.event_flags) &&
-			!is_value_equal(current_val, sent_val, value_type))
-		rc = 1;
-	else if (is_lower_flag_set(event.event_flags) &&
-			is_lower_than_threshold(current_val,
+	if (event.event_flags) {
+		switch (event.event_flags) {
+		case KNOT_EVT_FLAG_CHANGE:
+			if (!is_value_equal(current_val, sent_val, value_type))
+				rc = 0;
+			break;
+		case KNOT_EVT_FLAG_LOWER_THRESHOLD:
+			if (!is_lower_than_threshold(current_val,
 				event.lower_limit, value_type))
-		rc = 1;
-	else if (is_upper_flag_set(event.event_flags) &&
-			is_higher_than_threshold(current_val,
+				rc = 0;
```

Comment:
what does rc means? And I think the number zero should be replaced by a variable with an explicit name